src/main/kotlin/com/supergrecko/questionbot/tools/Arguments.kt

```diff
@@ -2,12 +2,39 @@ package com.supergrecko.questionbot.tools
 
 open class Arguments(val args: List<Any?>) {
 
-    inline fun <reified T> asType(index: Int): T? {
-        return args.getOrNull(index) as? T
+    /**
+     * Get an item from the argument list as type T.
+     *
+     * If item is not T or is null then `onError` will be ran and the final
+     * cast will always fail.
+     *
+     * @param index the index in the list to grab from
+     * @param onError the error handler if item isn't there
+     */
+    inline fun <reified T> asType(index: Int, onError: () -> Unit = {}): T {
+        val arg = args.getOrNull(index)
+
+        if (arg != null && arg is T) {
+            return arg
+        }
+
+        onError()
+        // This will always fail which is why default() should deal with errors
+        return arg as T
```

Comment:
I'm not a fan of this one, but it is how we trick the compiler to believe T is always returned. This will always throw an invalid cast. One should use the onError to deal with the error.

Open to suggestions on how to fix this